zio-config/src/main/scala/zio/config/actions/ConfigDocs.scala

```diff
@@ -1,49 +1,100 @@
 package zio.config.actions
 
 import zio.config.ConfigDescriptor
-import zio.config.actions.ConfigDocs.KeyDescription
 
-final case class ConfigDocs(configKeysAndDescription: List[KeyDescription], or: Option[ConfigDocs])
+sealed trait ConfigDocs
 
 // Man page
 object ConfigDocs {
-  final case class KeyDescription(path: String, list: List[String])
 
-  final def docs[A](config: ConfigDescriptor[A]): ConfigDocs = {
+  final case class Empty()                                  extends ConfigDocs
+  final case class Leaf(value: KeyDescription)              extends ConfigDocs
+  final case class And(left: ConfigDocs, right: ConfigDocs) extends ConfigDocs
+  final case class Or(left: ConfigDocs, right: ConfigDocs)  extends ConfigDocs
+
+  final case class KeyDescription(path: String, value: Option[String], docs: List[String])
+
+  final def createDoc[A](config: ConfigDescriptor[A], value: Option[A]): ConfigDocs = {
     def loop[B](
-      acc: List[String],
+      accumulatedDoc: List[String],
       previousDescription: String,
       config: ConfigDescriptor[B],
-      desc: ConfigDocs
+      desc: ConfigDocs,
+      value: Option[B]
```

Comment:
is this a trick to avoid `asInstanceOf` in case of `Optional`?

Reply:
No. The asInstanceOf was existing for no reason. The option is because for a given key user may or may not have passed a value.